internal/driver/credentials.go

```
@@ -103,13 +104,51 @@ func (d *Driver) tryGetCredentialsForDevice(device models.Device) (Credentials,
 		d.lc.Warnf("Device %s is missing MAC Address, using default secret name", device.Name)
 	}
 
-	credentials, edgexErr := d.tryGetCredentials(secretName)
+	credentials, edgexErr := d.tryGetCredentialsInternal(secretName)
 	if edgexErr != nil {
 		d.lc.Errorf("Failed to retrieve credentials for the secret name %s: %s", secretName, edgexErr.Error())
 		return Credentials{}, errors.NewCommonEdgeX(errors.KindServerError, "failed to get credentials", edgexErr)
 	}
 
-	d.lc.Debugf("Found credentials from secret name %s for device %s", secretName, device.Name)
-
 	return credentials, nil
 }
+
+func (d *Driver) secretUpdated(secretName string) {
+	d.lc.Infof("Secret updated callback called for secretName %s", secretName)
+
+	d.credsCacheMu.Lock()
+	// remove the cache entry for this secret
+	delete(d.credsCache, secretName)
+	d.credsCacheMu.Unlock()
+	_, _ = d.getCredentials(secretName) // update cached data
```

Why ignoring the error? Should at least log it and maybe not continue?